streams/jvm/src/test/scala/zio/stream/StreamSpec.scala

```diff
@@ -1456,4 +1460,39 @@ class ZStreamSpec(implicit ee: org.specs2.concurrent.ExecutionEnv)
       .either
       .map(_ must_=== Left("Ouch"))
   }
+
+  private def zipWithLatest = unsafeRun {
+    import java.util.concurrent.CountDownLatch
+    val latch1 = new CountDownLatch(2)
+    val latch2 = new CountDownLatch(2)
+    val latch3 = new CountDownLatch(1)
+    // Merely awaiting the latch does not guarantee runCollect will collect elements in the emission order.
```

Comment:
I don't know why awaiting the latch still does not guarantee runCollect will collect elements in the emission order. So I added a sleep here. Any better ideas?